### PR TITLE
Dave Vazquez - WEBPT5 - JavaScript-IV

### PR DESCRIPTION
**Pull request by dave-vazquez**

### CODE REVIEW

File: assignments/lambda-classes.js [hunk 1]
```diff
@@ -1 +1,315 @@
...
+                student.grade = 0;
+            }
+            else if(student.grade > 100){
+                student.grade = 100;
+            }
+
+            console.log(`${this.name} applied ${points} to ${student.name}'s grade.`);
+            console.log(`What a ${points <= 0 ? 'jerk...' : 'totally awesome instructor!'}`);
+            console.log(`${student.name}'s grade is now ${student.grade}`);
+            console.log(`${student.name} is ${student.grade <= 70 ? 'sad...' : 'elated!'}\n\n`);
```
Comment (Mister-Corn) [inline assignments/lambda-classes.js:50]: Your ternary game is strong, my dude :fire:

File: assignments/lambda-classes.js [hunk 2]
```diff
@@ -1 +1,315 @@
...
+    gender: 'male',
+    favLanguage: 'JavaScript',
+    specialty: 'Front-end',
+    catchPhrase: `catchphrase` // does cam have a catchphrase?? i don't know...
+});
+
+// my first cs professor, I never would have pursued software
+// engineering if it wasn't for her, forever in debt to her
+const darcy = new Instructor({
+    name: 'Darcy',
```
Comment (Mister-Corn) [inline assignments/lambda-classes.js:124]: Thank you, Darcy, for leading Dave into software engineering!

File: assignments/lambda-classes.js [hunk 3]
```diff
@@ -1 +1,315 @@
...
+}
+
+const cam = new Instructor({
+    name: 'Cam',
+    location: 'Utah',
+    age: 30,
+    gender: 'male',
+    favLanguage: 'JavaScript',
+    specialty: 'Front-end',
+    catchPhrase: `catchphrase` // does cam have a catchphrase?? i don't know...
```
Comment (Mister-Corn) [inline assignments/lambda-classes.js:118]: Crippling self-doubt? That's not really a catchphrase though... :thinking:

File: assignments/lambda-classes.js [hunk 4]
```diff
@@ -1 +1,315 @@
...
+        'Javascript',
+        'Java',
+        'English',
+        'Mathematics'
+    ],
+    grade: 1,
+    graduated: false
+});
+
+// my brother (not actually in lambda school, he recently graduated from CodeSmith)
```
Comment (Mister-Corn) [inline assignments/lambda-classes.js:151]: He went to the campus in New York, right? I've been to the one in LA a couple of times.

File: assignments/lambda-classes.js [hunk 5]
```diff
@@ -1 +1,315 @@
...
+        'Politics'
+    ],
+    grade: 1,
+    graduated: false
+});
+
+const ron = new ProjectManager({
+    name: 'Ron',
+    location: 'California',
+    age: 30, // no idea how old you are
```
Comment (Mister-Corn) [inline assignments/lambda-classes.js:173]: You can put `undefined` here :grin:

File: assignments/lambda-classes.js [hunk 6]
```diff
@@ -1 +1,315 @@
...
+
+    ryan.graduate();
+    dave.graduate();
+
+    console.log('\n*****************************************************\n\n');
+
+    if(dave.graduated && ryan.graduated) {
+        console.log(`\n\n${dave.name} and ${ryan.name} both graduated!!!\n\n`)
+        clearInterval(sprint);
+    }
```
Comment (Mister-Corn) [inline assignments/lambda-classes.js:281]: Very well done here!

File: assignments/prototype-refactor.js [hunk 7]
```diff
@@ -7,3 +7,203 @@ Prototype Refactor
...
+        super(attr);
+        this.team = attr.team;
+        this.weapons = attr.weapons;
+        this.language = attr.language;
+    }
+
+    greet() {
+        return `${this.name} offers a greeting in ${this.language}`;
+    }
+}
```
Comment (Mister-Corn) [inline assignments/prototype-refactor.js:45]: Great formatting, proper inheritance. All the MVP console logs and your stretch work exactly like before.

**Review by Mister-Corn — APPROVED** [review]
## Project Objectives

### `prototype-refactor`

* [x] Copy the work from JavaScript-III.
* [x] **Convert** all of your constructors into ES6 Classes using the `class` and `extends` keywords. You should be able to run your same logs and they should build out the proper expected behaviors.

### `lambda-classes`

* Develop the following classes.
  * [x] Person
  * [x] Instructor
  * [x] Student
  * [x] Project Manager

## Comments

Amazing work, Dave! :100: Since you really dug into prototypes and constructors, I felt this assignment would come easily for you, since JS class syntax is just a prettier version of the same thing. 

Overall, your JS skills are pretty strong. You're using ternaries and `++` well, but more importantly, the fun stuff you do adds, not detracts, to the readability of the code. Your code cannon stretch from the other project also shows you can do DOM manipulation already, so the next two weeks might be a reinforcement of what you probably already know. If you feel the next two weeks come more easily than expected, try taking a poke at React.

Again if you have any questions, you are welcome to reach out to me any time. Keep up the amazing work!

File: assignments/lambda-classes.js (hunk 4 above)
Comment (dave-vazquez) [reply assignments/lambda-classes.js:151]: Yep! I actually decided to enroll in Lambda after seeing his success at CodeSmith. He was the Lead Engineering Fellow there (basically what Chaya is to the PM's in Lambda). Though he recently quit and is off looking for an actual developer job now.

**Pull request merged**